src/data/rat/floor.lean

```diff
@@ -149,4 +149,46 @@ begin
   rwa [q_inv_eq, this.left, this.right, q_num_abs_eq_q_num, mul_comm] at q_inv_num_denom_ineq
 end
 
+namespace norm_num
+open tactic
+
+/-- A norm_num extension for `int.floor`, `int.ceil`, and `int.fract`. -/
+@[norm_num] meta def eval_floor_ceil : expr → tactic (expr × expr)
+| e@`(@int.floor %%R %%inst_1 %%inst_2 %%x) := do
+  q ← x.to_rat,
+  let z := int.floor q,
+  let ze := `(z),
+  t ← to_expr ``(%%e = %%ze),
+  ((), p) ← tactic.solve_aux t (do
+    tactic.mk_mapp `int.floor_eq_iff [some R, some inst_1, some inst_2] >>= tactic.rewrite_target,
+    tactic.interactive.norm_num [] (interactive.loc.ns [none])),
+  p ← instantiate_mvars p,
+  pure (ze, p)
+| e@`(@int.ceil %%R %%inst_1 %%inst_2 %%x) := do
+  q ← x.to_rat,
+  let z := int.ceil q,
+  let ze := `(z),
+  t ← to_expr ``(%%e = %%ze),
+  ((), p) ← tactic.solve_aux t (do
+    tactic.mk_mapp `int.ceil_eq_iff [some R, some inst_1, some inst_2] >>= tactic.rewrite_target,
+    `[norm_num]),
```

Comment:
Should the tactic input `solve_aux` end with a `done`, to make sure that it proves `t`, or is it unnecessary in this case, since this is an extension anyway?